Sources/Sapi.php

```diff
@@ -0,0 +1,318 @@
+<?php
+
+/**
+ * Simple Machines Forum (SMF)
+ *
+ * @package SMF
+ * @author Simple Machines https://www.simplemachines.org
+ * @copyright 2024 Simple Machines and individual contributors
+ * @license https://www.simplemachines.org/about/smf/license.php BSD
+ *
+ * @version 3.0 Alpha 1
+ */
+
+declare(strict_types=1);
+
+namespace SMF;
+
+/**
+ * Server Application Programming Interface.
+ *
+ * This file handles all interaction between the application (SMF) and the server.
+ *
+ * Not all APIs have are functional on all server interfaces.  If this is the case
+ * the API call will do nothing.
+ *
+ * Some functions may not be defined or definable in the documentation as they are
+ * bundled by 3rd party processes, such as apache_* functions.  To surpress IDE
+ * warnings, we use 'suppress PHP0417'.
+ */
+class Sapi
+{
+	/*********************
+	 * Internal properties
+	 *********************/
+
+	/**
+	 * @var array
+	 *
+	 * Server Map.
+	 */
+	protected array $server_software = [
+		self::SERVER_IIS => 'Microsoft-IIS',
+		self::SERVER_APACHE => 'Apache',
+		self::SERVER_LITESPEED => 'LiteSpeed',
+		self::SERVER_LIGHTTPD => 'lighttpd',
+		self::SERVER_NGINX => 'nginx'
+	];
+
+	/*****************
+	 * Class constants
+	 *****************/
+
+	public const SERVER_IIS = 'iis';
+	public const SERVER_APACHE = 'apache';
+	public const SERVER_LITESPEED = 'litespeed';
+	public const SERVER_LIGHTTPD = 'lighttpd';
+	public const SERVER_NGINX = 'nginx';
+
+	public const OS_WINDOWS = 'Windows';
+	public const OS_MAC = 'Darwin';
+	public const OS_LINUX = 'Linux';
+
+	/***********************
+	 * Public static methods
+	 ***********************/
+
+	/**
+	 * Checks if the server is running a specific server software.
+	 * 
+	 * @param string|string[] $server The server software as defined in our constants.
+	 * @return bool True if we are running the requested software, false otherwise.
+	 */
+	public static function isSoftware(string|array $server): bool
+	{
+		$servers = (array) $server;
+
+		foreach ($servers as $server) {
+			if (
+				isset($_SERVER['SERVER_SOFTWARE'])
+				&& strpos($_SERVER['SERVER_SOFTWARE'], self::$server_software[$server] ?? $server) !== false	
+			) {
+				return true;
+			}	
+		}
+
+		return false;
+	}
+
+	/**
+	 * Checks if the server is running a specific operating system.
+	 * 
+	 * @param string|string[] $os The os as defined in our constants
+	 * @return bool True if we are running the requested os, false otherwise.
+	 */
+	public static function isOS(string|array $os): bool
+	{
+		$oses = (array) $os;
+
+		/** 
+		 * Technically we could simplify this down using PHP_OS_FAMILY,
+		 * but to ensure backwards compatibility, we won't yet.
+		 */
+		foreach ($oses as $os) {
```

Comment:
multiline comments start with `/*`; phpdoc starts with `/**`

Reply:
There were a bunch of things in this PR that made PHP-CS-Fixer sad. Apparently the GitHub action didn't catch them for some reason. I push a commit to fix them all. However, we clearly need to fix something in the GitHub action so that it flags these problems like it is supposed to.

Reply:
And I have a habit of not checking php-cs-fixer before sending.  But I don't see any logic in our github actions that runs cs fixer.  Just a linter

Reply:
The problem is that the linter should have flagged the formatting issues and therefore failed the check. It didn't do that.